service/policy/db/grant_mappings.go

```diff
@@ -0,0 +1,111 @@
+package db
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/opentdf/platform/protocol/go/policy"
+)
+
+func mapAlgorithmToKasPublicKeyAlg(alg policy.Algorithm) policy.KasPublicKeyAlgEnum {
+	switch alg {
+	case policy.Algorithm_ALGORITHM_RSA_2048:
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_RSA_2048
+	case policy.Algorithm_ALGORITHM_RSA_4096:
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_RSA_4096
+	case policy.Algorithm_ALGORITHM_EC_P256: // ALGORITHM_EC_P256 is an alias
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_EC_SECP256R1
+	case policy.Algorithm_ALGORITHM_EC_P384: // ALGORITHM_EC_P384 is an alias
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_EC_SECP384R1
+	case policy.Algorithm_ALGORITHM_EC_P521: // ALGORITHM_EC_P521 is an alias
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_EC_SECP521R1
+	case policy.Algorithm_ALGORITHM_UNSPECIFIED:
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_UNSPECIFIED
+	default:
+		return policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_UNSPECIFIED
+	}
+}
+
+func mapKasKeysToGrants(keys []*policy.KasKey, existingGrants []*policy.KeyAccessServer) ([]*policy.KeyAccessServer, error) {
+	kasMap := make(map[string]*policy.KeyAccessServer)
+
+	// Populate the map with existing grants
+	for _, grant := range existingGrants {
```

Comment:
why might these be nil?

Reply:
They shouldn't be nil or even be missing a uri but we don't want to add them to the map if that ends up being the case some how.